common/txmgr/types/stuck_tx_detector.go

```diff
@@ -0,0 +1,21 @@
+package types
+
+import (
+	"context"
+
+	feetypes "github.com/smartcontractkit/chainlink/v2/common/fee/types"
+	"github.com/smartcontractkit/chainlink/v2/common/types"
+)
+
+type StuckTxDetector[
+	CHAIN_ID types.ID, // CHAIN_ID - chain id type
+	ADDR types.Hashable, // ADDR - chain address type
+	TX_HASH, BLOCK_HASH types.Hashable, // various chain hash types
+	SEQ types.Sequence, // SEQ - chain sequence type (nonce, utxo, etc)
+	FEE feetypes.Fee, // FEE - chain fee type
+] interface {
+	DetectStuckTransactions(ctx context.Context, enabledAddresses []ADDR, blockNum int64) ([]Tx[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE], error)
+	LoadPurgeBlockNumMap(ctx context.Context, addresses []ADDR) error
```

Comment:
I kept the name as Load since this method is called even after the `StuckTxDetector` is initialized. I feel like initialized implies that it's only called once.